### ISSUE TITLE
c2rust-ast-exporter build.rs: "no template named 'index_sequence' in namespace 'std'", etc.

### ISSUE THREAD

**Issue by trentj** [immunant/c2rust#439]
I am trying to compile c2rust from crates.io, and everything seems to work but `c2rust-ast-exporter`. The build process emits a list of C++ compiler errors:

```
  In file included from /home/trentj/.cargo/registry/src/github.com-1ecc6299db9ec823/c2rust-ast-exporter-0.16.0/src/AstExporter.cpp:11:
  In file included from /home/trentj/.conda/envs/lando/include/llvm/Support/CommandLine.h:22:
  In file included from /home/trentj/.conda/envs/lando/include/llvm/ADT/ArrayRef.h:15:
  /home/trentj/.conda/envs/lando/include/llvm/ADT/STLExtras.h:559:49: error: no template named 'index_sequence' in namespace 'std'
    template <size_t... Ns> value_type deref(std::index_sequence<Ns...>) const {
                                             ~~~~~^
  /home/trentj/.conda/envs/lando/include/llvm/ADT/STLExtras.h:564:36: error: no template named 'index_sequence' in namespace 'std'
    decltype(iterators) tup_inc(std::index_sequence<Ns...>) const {
                                ~~~~~^
```
This goes on for a while and hits the error limit; there's some other errors in there too. [Full `cargo install` output.](https://github.com/immunant/c2rust/files/8880043/c2rust_output.txt) But since these errors originate not in c2rust itself but in LLVM, I think it must be a configuration issue.

I'm not much of a C++ programmer. As best I can figure, this file is supposed to be compiled with a different language version or compiler version, or perhaps a different version of LLVM itself. I don't have administrator privileges on this machine, so I have installed LLVM and Clang 10.0.1 from conda. There is no system version.

How can I understand the errors and narrow down the cause of this build problem?

(I also tried installing from master with `cargo install --git`, to the same effect.)

**Comment by aneksteind** [comment]
I think (but am not totally sure) that this has to do with CMake and Conda [not playing nicely together](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#an-aside-on-cmake-and-sysroots). So I think the wrong sysroot is being used and it's pointing to header files that are for the C++11 (perhaps in `/usr/include`) and not the llvm C++14 headers

**Comment by trentj** [comment]
If I build with `VERBOSE=1`, this is the command line CMake reports as it compiles AstExporter.cpp:

```
  /home/trentj/.conda/envs/lando/bin/clang++ -DCLANG_BIN_PATH=\"/home/trentj/.conda/envs/lando/bin\" -DCLANG_LIBDIR_SUFFIX=\"\" -DCLANG_VERSION_STRING=\"10.0.1\" -I/tmp/cargo-installSWXjxd/release/build/c2rust-ast-exporter-9223552c36ccbf68/out/build/include -I/home/trentj/.conda/envs/lando/include -ffunction-sections -fdata-sections -fPIC --target=x86_64-unknown-linux-gnu -fno-rtti -O3 -DNDEBUG   -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -MD -MT CMakeFiles/clangAstExporter.dir/AstExporter.cpp.o -MF CMakeFiles/clangAstExporter.dir/AstExporter.cpp.o.d -o CMakeFiles/clangAstExporter.dir/AstExporter.cpp.o -c /home/trentj/.cargo/registry/src/github.com-1ecc6299db9ec823/c2rust-ast-exporter-0.16.0/src/AstExporter.cpp
```

I tried with `CXXFLAGS="-std=c++14"` (and `c++17`, `c++20` for good measure). The command line (as reported by CMake) **does** change to include the flag, but the compiler output is no different.

```
$ /home/trentj/.conda/envs/lando/bin/clang --version
clang version 10.0.1
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/trentj/.conda/envs/lando/bin
```

**Comment by fw-immunant** [comment]
Changing the explicitly-requested C++ std version does make it seem like this is a sysroot issue rather than a `-std=...` issue. See if the approach with `-DCMAKE_TOOLCHAIN_FILE="${RECIPE_DIR}/cross-linux.cmake` and [`cross-linux.cmake`](https://github.com/AnacondaRecipes/libnetcdf-feedstock/blob/master/recipe/cross-linux.cmake) mentioned in @aneksteind's link works.

**Comment by trentj** [comment]
This CMake angle looks promising. When passing `-stdlib=libc++` I get something similar to the error described there, which I think makes sense. It's getting late here so I'll have to try the fix on Monday.

**Comment by trentj** [comment]
It seems I misconfigured the local environment somehow, and the CMake issue was confusing my attempts to resolve it.

To fix it properly, I think I would need to modify build.rs to pass the appropriate flags to cmake, but before that I would need to fix my sysroot environment, and I've already spent a day+ fiddling with it. I only really need to run c2rust a couple times, so I'm just going to do it on another computer with a newer toolchain.

I feel closing with no action is warranted since the issue here is primarily my system configuration. Although I suppose it's arguable whether there should be a workaround for this specific situation in build.rs.
